rfcs/2024-08-21-motion-presets.md

```diff
@@ -0,0 +1,523 @@
+
+## POCs
+
+### Hardware Accelarated Motion using Framer Motion
+
+https://github.com/user-attachments/assets/5aa7ba3f-666e-449c-ae1d-821f93c12962
+
+<p align="right">
+  <a href="https://codesandbox.io/p/sandbox/framer-motion-enter-animation-forked-rk3tvd?file=%2Fsrc%2FApp.tsx%3A27%2C28"><img src="https://codesandbox.io/static/img/play-codesandbox.svg" /></a>
+</p>
+
+### Hardware Accelarated Motion using GSAP
+
+> [!NOTE]
+>
+> While GSAP does offload some work to hardware using CSS, it still requires javascript to work and stops working if JS thread is blocked
+
+https://github.com/user-attachments/assets/57593b19-3a19-4713-b768-9bcbf69097ee
+
+<p align="right">
+<a href="https://codesandbox.io/p/sandbox/framer-motion-hardware-accelarated-animations-forked-7qjyg9">
+  <img alt="Edit GSAP: Hardware Accelarated Animations " src="https://codesandbox.io/static/img/play-codesandbox.svg">
+</a>
+</p>
+
+### Morph / Layout Animations with Framer Motion
+
+https://github.com/user-attachments/assets/ad3d4a23-c3b9-4980-a051-a0f44e7224dc
+
+<p align="right">
+  <a href="https://codesandbox.io/p/sandbox/framer-motion-hardware-accelarated-animations-forked-q674t5">
+  <img alt="Edit Framer Motion: Morph Animations" src="https://codesandbox.io/static/img/play-codesandbox.svg">
+</a>
+</p>
+
+### Page Transitions with Framer Motion and React Router
+
+Goal of the POC was to make sure if its possible to animate some part of the page while keeping the other part of the page stable. It was success with framer motion
+
+https://github.com/user-attachments/assets/b53779ec-55ec-4ef4-bdca-60b2e46cb3ae
+
+<p align="right">
+  <a href="https://codesandbox.io/p/sandbox/inspiring-stallman-2wn2v5">
+  <img alt="Edit Framer Motion: Morph Animations" src="https://codesandbox.io/static/img/play-codesandbox.svg">
+</a>
+</p>
+
+### Other POCs
+
+- [Framer Motion Layout Animations with Blade Components](https://stackblitz.com/edit/framer-motion-blade-poc?file=App.tsx,Logger.tsx)
+- [Enhancer Component POC to check API feasibility](https://stackblitz.com/edit/enhancer-component-poc?file=App.tsx)
+
+# Drawbacks/Constraints
+
+- Framer Motion as a library will be introduced in customer projects which might increase their bundle size
+
+# Alternatives
+
+- Alternative libraries and native CSS solution is compared in [Library Comparison Section](#library-comparison)
+- Other alternative is to let consumers do animations
+  - Since there are less high level primitives available and it has led to inconsistent motion across products, we prefer to simplify building animations while giving out consistency
+
+# Adoption strategy
+
+- We plan to target 1 project this quarter (Q2) to get motion adopted
+- The new projects that are built, should be built with motion presets on design and dev
+- The earlier project that we have should use motion presets when they redesign / revamp
+
+# How do we educate people?
```

Comment:
lets add how will we get this implemented on Figma  for easing the hand-off as well here?

Reply:
Yup I'll update it here. We're still figuring out the right approach. We discussed few things yesterday, RK will just confirm it once if things are working fine or not